### ISSUE TITLE
wineasio.dll failed to register/load on Wine-6.0-RC5 (patched)

### ISSUE THREAD

**Issue by MosesKim84** [wineasio/wineasio#20]
Hello, 

I have been using wineasio under wine-stable (5.0.3) on Ubuntu 20.10 x64. I have been using it with Reaper 6.19 and have successfully had it register without much incident, and also working with Jack2 as well. 

There have been several new libraries released under the new Wine 6.0 release candidates, along with a patch I need to run Presonus Studio One 5 Professional. I have compiled my own wine and wine64 and have been able to run Studio One with the latest patch. 

Unfortunately I cannot get wineasio.dll to register using: 

_regsvr32 wineasio.dll 
wine64 regsvr32 wineasio.dll_

with the error that it failed to load the DLL. I was able to get it to register using: 

_regsvr32 /usr/lib/i386-linx-gnu/wine/wineasio.dll.so
wine64 regsvr32 /usr/lib/x86_x64-linux-gnu/wine/wineasio.dll.so_

Unfortunately even with it registering, wineasio will not work under Reaper 6.19 or Studio One 5. An option for "wineasio"  under audio devices appears for both programs, but when I select it, an error comes up saying it failed to load the ASIO device and the port selection (in/out) is empty. When I debug both programs, it gives the same error: "failed to load wineasio.dll." I have compiled my own versions of wineasio from the latest source, and have tried using the kxstudio repos with no improvement.

I have also tried registering it using the WINEDLLPATH=/usr/lib64/wine and placing wineasio.dll.so in that path. It will successfully register as well, but it gives the same error when I try to select the ASIO device in the programs. 

When I revert back to Wine 5.0.3, the DLLs register without any additional specification (using only "_regsvr32 wineasio.dll; wine64 regsvr32 wineasio.dll"_) and work fine with jack. Unfortunately I cannot use Studio One on 5.0.3 due to multiple libraries being missing that were added in the Wine 5.0 development cycle. What's the best way to evaluate this further and get a potential fix going for wineasio or wine? Thanks for your help. 

-Moses

**Comment by falkTX** [comment]
You can try to enable all wine debug messages, then start the windows application via CLI so you can catch errors on the exact moment you try to load wineasio.

see https://www.winehq.org/docs/wineusr-guide/environment-variables

**Comment by EdwardSnowmen** [comment]
I ended up in the same rabbit hole you did after installing wine-6.0-rc6 in ubuntu 20.04.1. What did the trick for me was simply:
`sudo cp /usr/lib/i386-linux-gnu/wine/wineasio.dll.so /opt/wine-devel/lib/wine/wineasio.dll.so && sudo cp /usr/lib/x86_64-linux-gnu/wine/wineasio.dll.so /opt/wine-devel/lib64/wine/wineasio.dll.so`
Then I could register the *.dll:s for my 32/64-bit prefix with:
`regsvr32 wineasio.dll && wine64 regsvr32 wineasio.dll`
Hopefully this works for you too!

**Comment by MosesKim84** [comment]
> I ended up in the same rabbit hole you did after installing wine-6.0-rc6 in ubuntu 20.04.1. What did the trick for me was simply:
> `sudo cp /usr/lib/i386-linux-gnu/wine/wineasio.dll.so /opt/wine-devel/lib/wine/wineasio.dll.so && sudo cp /usr/lib/x86_64-linux-gnu/wine/wineasio.dll.so /opt/wine-devel/lib64/wine/wineasio.dll.so`
> Then I could register the *.dll:s for my 32/64-bit prefix with:
> `regsvr32 wineasio.dll && wine64 regsvr32 wineasio.dll`
> Hopefully this works for you too!

This did work for me in getting the unpatched RC-6 in /usr/bin/wine to finally! Thanks so much, that's a big step in the right direction. Also of note, I have /opt/wine-staging/ instead of /opt/wine-devel/ due to the package that I used. Using reaper with wine works as well, thanks!!

Now to the next issue: I'm using a patched version of wine-RC6 to get Studio One 5 to work. For some reason, the patched version of wine RC-6 that I compiled installs its binaries to /usr/local/bin/, and when I try using regsvr32 using that patched wine from that directory, it still fails. It seems like this patched version of wine is having trouble accessing wineasio.dll from these locations. I'm not sure how to specify this further. Thanks for everyone replies though
